specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/mfe.json

```diff
@@ -11882,6 +12010,58 @@
       "x-ms-discriminator-value": "Bandit",
       "additionalProperties": false
     },
+    "BaseEnvironmentId": {
+      "description": "Base environment type.",
+      "required": [
+        "resourceId"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseEnvironmentSource"
+        }
+      ],
+      "properties": {
+        "resourceId": {
+          "description": "[Required] Resource id accepting ArmId or AzureMlId.",
+          "minLength": 1,
+          "pattern": "[a-zA-Z0-9_]",
+          "type": "string"
```

Comment:
>"type": "string" [](http://example.com/codeflow?start=10&length=16)

Consider using format arm-id: and [optionally] a set of allowed resource types. Comment applies throughout.

Reply:
This value accepts both arm-id and AzureMLId.